tv_apps/smart-system/js/app_transition_controller.js

```diff
@@ -170,18 +171,35 @@
     }
   };
 
-
   AppTransitionController.prototype._do_opening =
     function atc_do_opening() {
+      this._waitingForLoad = false;
+      var animation = this.getAnimationName('open');
+      var color = (this.app.name === 'Apps' ||
+                   this.app.name === 'Dashboard') ? '#E0E0E0' : 'black';
+      if (animation === 'invoked') {
+        Animations.createCircleAnimation( this.app.element.parentNode, color)
+          .play('grow', function() {
+            // show the homescreen's fadeoverlay so we will not see the system
+            // background during fast-fade-in transition
+            homescreenWindowManager.getHomescreen().fadeOut();
+            homescreenWindowManager.getHomescreen().showFadeOverlay(color);
+            this.app.element.classList.add('fast-fade-in');
+            // the time out delay is the same as fast-fade-in
+            setTimeout(function() {
+              this.app.broadcast('openingtimeout');
+            }.bind(this), 300);
```

Comment:
Please add slowTransition mode, like:
```
AppWindowManager.slowTransition ? this.SLOW_TRANSITION_TIMEOUT :
                                                             this.INVOKE_TRANSITION_TIMEOUT;
```

Please note the this.INVOKE_TRANSITION_TIMEOUT should be declared by yourself.